src/test/java/com/example/javaOjt/controllers/AuthorControllerTest.java

```diff
@@ -265,4 +266,80 @@ void getAuthor_invalidWithBooks() throws Exception {
     Mockito.verify(authorService, Mockito.never())
         .getAuthorById(Mockito.anyInt(), Mockito.anyBoolean());
   }
+
+  @Test
+  void postAuthor() throws Exception {
+    String requestBody = "{\"name\": \"三島由紀夫\"}";
+    Author savedAuthor = new Author();
+    savedAuthor.setId(1);
+    savedAuthor.setName("三島由紀夫");
+
+    // Mocking the service response
+    Mockito.when(authorService.postAuthor(Mockito.any(String.class)))
+        .thenReturn(new GetAuthorResponse(savedAuthor));
+
+    // Perform the POST request with the POST request body
+    mockMvc.perform(
+            MockMvcRequestBuilders.post(AUTHOR_BASE_URL).contentType(MediaType.APPLICATION_JSON)
+                .content(requestBody))
+        .andExpect(MockMvcResultMatchers.status().isCreated())
+        .andExpect(MockMvcResultMatchers.header().string("Location", "/authors/1"))
+        .andExpect(MockMvcResultMatchers.content().string(""))
+        .andReturn();
```

Comment:
この validation が間違ったのでレスポンスボディのバグが発覚できなくなっちゃいましたね。

```suggestion
.andExpect(MockMvcResultMatchers.content().contentType(MediaType.APPLICATION_JSON))
        .andExpect(
            MockMvcResultMatchers.jsonPath("$.id").value(1))
        .andExpect(MockMvcResultMatchers.jsonPath("$.name").value("三島由紀夫"))
        .andReturn();
```